project/apps/identity_provider/authentication.py

```diff
@@ -0,0 +1,34 @@
+from rest_framework.authentication import TokenAuthentication
+from rest_framework import exceptions
+from django.utils.translation import gettext_lazy as _
+from django.core.exceptions import ValidationError
+from apps.identity_provider.models import ApiKey
+from datetime import datetime, timezone
+
+class DjamTokenAuthentication(TokenAuthentication):
+    
+    # Set the ApiKey model instead of the defult Token model 
+    model = ApiKey
+
+    def authenticate_credentials(self, key):
+        # Get the our custom ApiKey model and not the default one
+        model = self.get_model()
+        try:
+            token = model.objects.select_related('user').get(key=key)
+        except model.DoesNotExist:
+            raise exceptions.AuthenticationFailed(_('Invalid token.'))
+        except ValidationError:
```

Comment:
hi @Gpetrak 
if you want to return the same exeption you can merge the two lines as follow:

```python
except (model.DoesNotExist, ValidationError):
    raise exceptions.AuthenticationFailed(_('Invalid token.'))
```